specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/privateLinkService.json

```diff
@@ -0,0 +1,618 @@
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "Private link service resource."
+    },
+    "PrivateLinkServiceProperties": {
+      "properties": {
+        "loadBalancerFrontendIPConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "./loadBalancer.json#/definitions/FrontendIPConfiguration"
+          },
+          "description": "An array of references to the load balancer IP configurations."
+        },
+        "ipConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkServiceIpConfiguration"
+          },
+          "description": "An array of references to the private link service IP configuration."
+        },
+        "networkInterfaces": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "./networkInterface.json#/definitions/NetworkInterface"
+          },
+          "description": "Gets an array of references to the network interfaces created for this private link service."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state of the private link service. Possible values are: 'Updating', 'Succeeded', and 'Failed'."
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          },
+          "description": "An array of list about connections to the private endpoint."
+        },
+        "visibility": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/ResourceSet"
+            }
+          ],
+          "description": "The visibility list of the private link service."
+        },
+        "autoApproval": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/ResourceSet"
+            }
+          ],
+          "description": "The auto-approval list of the private link service."
+        },
+        "fqdns": {
+          "type": "array",
+          "items": {
+            "type":"string"
+          },
+          "description": "The list of Fqdn."
+        },
+        "alias": {
+          "readOnly": true,
+          "type": "string",
+          "description":"The alias of the private link service."
+        }
+      },
+      "description": "Properties of the private link service."
+    },
+    "ResourceSet": {
+      "properties": {
+        "subscriptions": {
+          "type": "array",
+          "items": {
+            "type":"string"
+          },
+          "description": "The list of subscriptions."
+        }
+      },
+      "description": "The base resource set for visibility and auto-approval."
+    },
+    "PrivateLinkServiceIpConfiguration": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateLinkServiceIpConfigurationProperties",
```

Comment:
@number213 This involves a service side change, if the service is open to integrating these changes at their end this would work, not otherwise
per offline discussions with @bowen0106 looks like the service has already been deployed, please advise them on whether they should fix the service too

Reply:
@dsgouda doesn't `x-ms-client-flatten` affect only SDKs? I don't get how server-side is involved here.
I've tried PUT'ing PrivateEndpoint object from .NET SDK after re-generating it with and without `x-ms-client-flatten`.

This is code without the flag
```csharp
var subnet = new Subnet()
{
    Id = @"/subscriptions/.../test-subnet"
};
var props = new PrivateEndpointProperties()
{
    Subnet = subnet
};
var endpoint = new PrivateEndpoint()
{
    Location = location,
    Properties = props
};
var putEndpointResponse = networkManagementClient.PrivateEndpoints.CreateOrUpdate(resourceGroupName, endpointName, endpoint);
```

This is what I see in Jarvis logs
```json
{
  "properties": {
    "subnet": {
      "id": "/subscriptions/.../subnets"
    }
  },
  "location": "eastus2"
}
```
---
This is code with `x-ms-client-flatten`
```csharp
var subnet = new Subnet()
{
    Id = @"/subscriptions/.../test-subnet"
};
var endpoint = new PrivateEndpoint()
{
    Location = location,
    Subnet = subnet
};
var putEndpointResponse = networkManagementClient.PrivateEndpoints.CreateOrUpdate(resourceGroupName, endpointName, endpoint);
```
And this is log in Jarvis
```json
{
  "properties": {
    "subnet": {
      "id": "/subscriptions/.../test-subnet"
    }
  },
  "location": "eastus2"
}
```

Reply:
@dsgouda as you can see, with or without `x-ms-client-flatten` SDK handles de-/serializing correctly

Reply:
@number213 I believe you're correct.
@bowen0106 Please refer the example and explanation [here](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-flatten)